shared/models_test.go

```diff
@@ -0,0 +1,46 @@
+// +build medium
+
+package shared_test
```

Comment:
Vain curiosity: Is there are reason this is using `shared_test` (i.e., black-box testing)? Most of our code just uses `foo` instead of `foo_test`.

Reply:
There was a circular dep; perhaps I should change back and solve it properly.

Reply:
Ok so to document the offline discussion:
As you mention, `shared_test` is a special-case is Golang that lets you test a package as though you're external to it.
We _were_ depending on `shared` from `shared/sharedtest`, which was circular if running tests from `shared`. I've removed that somewhat unnecessary dependency.